src/main/java/uk/gov/hmcts/reform/wacaseeventhandler/handlers/WarningTaskHandler.java

```diff
@@ -2,24 +2,86 @@
+import uk.gov.hmcts.reform.wacaseeventhandler.domain.handlers.cancellationtask.CancellationCorrelationKeys;
+import uk.gov.hmcts.reform.wacaseeventhandler.domain.handlers.cancellationtask.CancellationEvaluateRequest;
+import uk.gov.hmcts.reform.wacaseeventhandler.domain.handlers.common.DmnStringValue;
+import uk.gov.hmcts.reform.wacaseeventhandler.domain.handlers.common.EvaluateDmnRequest;
 import uk.gov.hmcts.reform.wacaseeventhandler.domain.handlers.common.EvaluateResponse;
 import uk.gov.hmcts.reform.wacaseeventhandler.domain.handlers.common.EventInformation;
+import uk.gov.hmcts.reform.wacaseeventhandler.domain.handlers.common.ProcessVariables;
+import uk.gov.hmcts.reform.wacaseeventhandler.domain.handlers.common.SendMessageRequest;
 import uk.gov.hmcts.reform.wacaseeventhandler.domain.handlers.warningtask.WarningEvaluateResponse;
 
-import java.util.Collections;
 import java.util.List;
 
+import static uk.gov.hmcts.reform.wacaseeventhandler.services.HandlerConstants.TASK_WARN;
+
 @Service
 @Order(2)
 public class WarningTaskHandler implements CaseEventHandler {
+
+    private final WorkflowApiClientToWarnTask workflowApiClientToWarnTask;
+
+    public WarningTaskHandler(WorkflowApiClientToWarnTask workflowApiClientToWarnTask) {
+        this.workflowApiClientToWarnTask = workflowApiClientToWarnTask;
+    }
+
     @Override
     public List<WarningEvaluateResponse> evaluateDmn(EventInformation eventInformation) {
-        // placeholder for the future cancellation process
-        return Collections.emptyList();
+        String tableKey = TASK_WARN.getTableKey(
+            eventInformation.getJurisdictionId(),
+            eventInformation.getCaseTypeId()
+        );
+
+        EvaluateDmnRequest<CancellationEvaluateRequest> requestParameters = getParameterRequest(
+            eventInformation.getPreviousStateId(),
+            eventInformation.getEventId(),
+            eventInformation.getNewStateId()
+        );
+
+        return workflowApiClientToWarnTask.evaluateDmn(tableKey, requestParameters).getResults();
+    }
+
+    private EvaluateDmnRequest<CancellationEvaluateRequest> getParameterRequest(
+        String previousStateId,
+        String eventId,
+        String newStateId
+    ) {
+        CancellationEvaluateRequest variables = new CancellationEvaluateRequest(
+            new DmnStringValue(eventId),
+            new DmnStringValue(newStateId),
+            new DmnStringValue(previousStateId)
+        );
+
+        return new EvaluateDmnRequest<>(variables);
     }
 
     @Override
     public void handle(List<? extends EvaluateResponse> results, EventInformation eventInformation) {
-        // empty for now
+        results.stream()
+            .filter(result -> result instanceof WarningEvaluateResponse)
+            .map(result -> (WarningEvaluateResponse) result)
+            .forEach(cancellationEvaluateResponse -> sendMessageToWarnTasksForGivenCorrelations(
```

Comment:
Yes, I understand what you say, @raghera . However, the forEach iterates over the rows resulting from evaluating the Cancellation DMN (the warning rule is part of this table as it is the cancellation rules, yes). Therefore, yes, we can create a new object and name it as you say, but it's going to look exactly the same as the CancellationEvaluateResponse . Just wanted to clarify it. Having said, I agree with Ravi's change since it makes it clearer.